js/search/search.js

```diff
@@ -45,17 +51,27 @@ class Search {
     /**
      * Constructor for the SymphonySearchEngine library
      * @param userId (for the index folder name)
+     * @param key
      */
-    constructor(userId) {
+    constructor(userId, key) {
         this.isInitialized = false;
         this.userId = userId;
-        this.startIndexingFromDate = (new Date().getTime() - SEARCH_PERIOD_SUBTRACTOR).toString();
-        this.indexFolderName = INDEX_PREFIX + '_' + userId + '_' + INDEX_VERSION;
+        this.key = key;
+        this.indexFolderName = INDEX_PREFIX + '_' + this.userId + '_' + INDEX_VERSION;
         this.dataFolder = INDEX_DATA_FOLDER;
         this.realTimeIndex = TEMP_REAL_TIME_INDEX;
         this.batchIndex = TEMP_BATCH_INDEX_FOLDER;
         this.messageData = [];
-        this.init();
+        this.crypto = new Crypto(userId, key);
+        this.decryptAndInit();
+    }
+
+    decryptAndInit() {
+        this.crypto.decryption().then(() => {
+            this.init();
+        }).catch(() => {
+            this.init();
```

Comment:
What's exactly going on here?

Reply:
Ensuring the search library is initialised even if the decryption fails